src/pickup.cpp

```diff
@@ -229,10 +234,12 @@ bool pick_one_up( const tripoint &pickup_target, item &newit, vehicle *veh,
             break;
         case WEAR:
             picked_up = !!u.wear_item( newit );
+            newit.handle_pickup_ownership( u );
```

Comment:
This needs to happen inside `wear_item()` or `on_wear()` as this isn't the only place in the code that causes a player to wear something from outside their inventory

Reply:
I knew I had a reason I moved it out, forgotten now.
It seems to fit nicely from on_pickup(), on_wield() and on_wear()
on_wield() takes a Player* class, but that should be no problem for handle_pickup_ownership() wanting a Character